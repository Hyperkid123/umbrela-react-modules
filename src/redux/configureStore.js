import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
