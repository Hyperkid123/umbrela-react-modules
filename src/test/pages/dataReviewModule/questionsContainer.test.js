import React from 'react'
import QuestionsContainer, {QuestionsContainer as Snapshot} from '../../../pages/dataReviewModule/questionsContainer';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Questions container component', () => {
  it('should render empty correclty' , () => {
      const tree = shallow(
        <Snapshot
          questionData={new Map()}
          translate={jest.fn()}
          match={{params: {questionId: 10}}}
          questions={{}}
          fetchQuestionIfNeeded={jest.fn()}
          ui={{
            chartType: 'PIE_CHART',
            hideChartlegend: false,
          }}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  const mockStore = configureStore();
  let initialState = {
    ui: {
      chartType: 'PIE_CHART',
      hideChartlegend: false,
    },
    data: {
      isFetching: false,
      questionData: {}
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<QuestionsContainer store={store}/>);
    expect(wrapper).toBeDefined();
  })

  it('should render with data correclty' , () => {
    const questionId = 1;
    const questionData = new Map();
    questionData.set(questionId, {});
      const tree = shallow(
        <Snapshot
          fetchQuestionIfNeeded={jest.fn()}
          questionData={questionData}
          translate={jest.fn()}
          match={{params: {questionId}}}
          questions={{"1": {
            title: 'foo',
            type: 'CloseQuestion'
          }}}
          ui={{
            chartType: 'PIE_CHART',
            hideChartlegend: false,
          }}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  it('should update data on question change', () => {
    const fetchQuestions = jest.fn();
    const questionId = 1;
    const questionData = new Map();
    questionData.set(questionId, {});
    const wrapper = shallow(
      <Snapshot
        fetchQuestionIfNeeded={fetchQuestions}
        questionData={questionData}
        translate={jest.fn()}
        match={{params: {questionId}}}
        questions={{"1": {
          title: 'foo',
          type: 'CloseQuestion'
        }}}
        ui={{
          chartType: 'PIE_CHART',
          hideChartlegend: false,
        }}
      />
    );
    wrapper.setProps({match:{params: {questionId: 2}}});
    expect(fetchQuestions.mock.calls.length).toEqual(2);
  })
});
