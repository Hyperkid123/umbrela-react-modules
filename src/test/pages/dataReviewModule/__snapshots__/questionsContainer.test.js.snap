// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions container component should render empty correclty 1`] = `
<styled.div
  grow={true}
  horizontalCenter={true}
  verticalCenter={true}
>
  <WithStyles(CircularProgress)
    size={80}
    thickness={5}
  />
</styled.div>
`;

exports[`Questions container component should render with data correclty 1`] = `
<div
  className="sc-gPEVay ienmZE"
>
  <styled.div
    transparent={true}
  >
    foo
  </styled.div>
  <styled.div
    transparent={true}
  >
    <Chart
      chartType="PIE_CHART"
      data={Object {}}
      question={
        Object {
          "title": "foo",
          "type": "CloseQuestion",
        }
      }
      questionType="CloseQuestion"
      showLegend={false}
    />
  </styled.div>
</div>
`;
